compiler/rustc_symbol_mangling/src/typeid/typeid_itanium_cxx_abi.rs

```diff
@@ -745,278 +747,278 @@ fn encode_ty<'tcx>(
-                        );
+            ty::Slice(ty0) => Ty::new_slice(self.tcx, self.fold_ty(*ty0)),
+
+            ty::Adt(adt_def, args) => {
+                if t.is_c_void(self.tcx) {
+                    Ty::new_unit(self.tcx)
+                } else if self.options.contains(TransformTyOptions::GENERALIZE_REPR_C)
+                    && adt_def.repr().c()
+                {
+                    Ty::new_adt(self.tcx, *adt_def, ty::List::empty())
+                } else if adt_def.repr().transparent()
+                    && adt_def.is_struct()
+                    && !self.parents.contains(&t)
+                {
+                    // Don't transform repr(transparent) types with an user-defined CFI encoding to
+                    // preserve the user-defined CFI encoding.
+                    if let Some(_) = self.tcx.get_attr(adt_def.did(), sym::cfi_encoding) {
+                        return t;
+                    }
+                    let variant = adt_def.non_enum_variant();
+                    let param_env = self.tcx.param_env(variant.def_id);
+                    let field = variant.fields.iter().find(|field| {
+                        let ty = self.tcx.type_of(field.did).instantiate_identity();
+                        let is_zst = self
+                            .tcx
+                            .layout_of(param_env.and(ty))
+                            .is_ok_and(|layout| layout.is_zst());
+                        !is_zst
+                    });
+                    if let Some(field) = field {
+                        let ty0 = self.tcx.type_of(field.did).instantiate(self.tcx, args);
+                        // Generalize any repr(transparent) user-defined type that is either a
+                        // pointer or reference, and either references itself or any other type that
+                        // contains or references itself, to avoid a reference cycle.
+
+                        // If the self reference is not through a pointer, for example, due
+                        // to using `PhantomData`, need to skip normalizing it if we hit it again.
+                        let mut ty = t;
+                        self.parents.push(ty);
+                        if ty0.is_any_ptr() && ty0.contains(ty) {
+                            self.options |= TransformTyOptions::GENERALIZE_POINTERS;
+                            ty = self.fold_ty(ty0);
+                            self.options &= !TransformTyOptions::GENERALIZE_POINTERS;
+                        } else {
+                            ty = self.fold_ty(ty0)
+                        }
+                        self.parents.pop();
+                        return ty;
                     } else {
-                        ty = transform_ty(tcx, ty0, parents, options);
+                        // Transform repr(transparent) types without non-ZST field into ()
+                        Ty::new_unit(self.tcx)
```

Comment:
Change all existing `Ty::new_unit`s to `self.tcx.types.unit`. Easier to read imo

Reply:
Done.